client/src/main/kotlin/io/spine/examples/pingh/client/MentionsFlow.kt

```diff
@@ -0,0 +1,207 @@
+            UserMentionsId::class.buildBy(session.value!!.username),
+            UserMentions::class
+        )
+        if (userMentions == null) {
+            return listOf()
+        }
+        return userMentions
+            .mentionList
+            .sortedByDescending { mention -> mention.whenMentioned.seconds }
+    }
+
+    /**
+     * Marks the mention as snoozed.
+     *
+     * @param id the identifier of the mention that is marked as snoozed.
+     */
+    public fun markMentionAsSnoozed(id: MentionId) {
+        markMentionAsSnoozed(id, settings.snoozeTime.value.value)
+    }
+
+    /**
+     * Marks the mention as snoozed.
+     *
+     * @param id the identifier of the mention that is marked as snoozed.
+     * @param snoozeTimeDuration the duration of mention snooze.
+     */
+    internal fun markMentionAsSnoozed(
+        id: MentionId,
+        snoozeTimeDuration: Duration
+    ) {
+        ensureLoggedIn()
+        val command = SnoozeMention::class.buildBy(
+            id,
+            currentTime().add(snoozeTimeDuration)
+        )
+        client.observeEventOnce(command.id, MentionSnoozed::class) {
+            mentions.value = mentions.value.setMentionStatus(id, MentionStatus.SNOOZED)
+        }
+        client.send(command)
+    }
+
+    /**
+     * Marks that the mention is read.
+     *
+     * @param id the identifier of the mention that is marked as read.
+     */
+    public fun markMentionAsRead(
+        id: MentionId
+    ) {
+        ensureLoggedIn()
+        val command = MarkMentionAsRead::class.buildBy(id)
+        client.observeEventOnce(command.id, MentionRead::class) {
+            mentions.value = mentions.value.setMentionStatus(id, MentionStatus.READ)
+        }
+        client.send(command)
+    }
+
+    /**
+     * Throws an `IllegalStateException` exception if the user is not logged in.
+     */
+    private fun ensureLoggedIn() {
+        check(session.value != null) { "The user is not logged in." }
+    }
+}
+
+/**
+ * List of `MentionsView`s.
+ */
+public typealias MentionsList = List<MentionView>
+
+/**
+ * Updates the status of the mention with the specified identifier to the new status.
+ *
+ * @param id the identifier of the mention which the status was changed.
+ * @param status new status of the mention.
+ */
+public fun MentionsList.setMentionStatus(
+    id: MentionId,
+    status: MentionStatus
+): MentionsList {
+    val idInList = this.indexOfFirst { it.id == id }
+    val updatedMention = this[idInList]
+        .toBuilder()
+        .setStatus(status)
+        .vBuild()
+    val newMentionsList = this.toMutableList()
+    newMentionsList[idInList] = updatedMention
+    return newMentionsList
+}
+
+/**
+ * Returns a `MentionsList` sorted such that unread mentions come first,
+ * followed by snoozed mentions, and read mentions last.
+ *
+ * Within each group, mentions are sorted by the time they were made.
+ */
+public fun MentionsList.sortByStatusAndWhenMentioned(): MentionsList =
```

Comment:
Do we ever sort them in another manner? If not, the method should be called like `sorted`.

Please start learning to shorten the method names. You have this problem throughout the PRs.